File: Python/Pascal's Triangle II.py
# Given an index k, return the kth row of the Pascal's triangle.

# For example, given k = 3,
# Return [1,3,3,1].

# Note:
# Could you optimize your algorithm to use only O(k) extra space?


class Solution:
    # @return a list of integers
    def getRow(self, rowIndex):
		result = [0 for i in xrange(rowIndex+2)]
		result[1] = 1
		for i in range(0,rowIndex):
			for j in range(rowIndex+1,-1,-1):
				result[j] = result[j-1] + result[j]
		result.pop(0)
		return result

File: Java/Pascal's Triangle II.java
class Pascal's Triangle II{
	public ArrayList<Integer> getRow(int rowIndex) {
		ArrayList<Integer> res = new ArrayList<Integer>();
		res.add(1);
		for(int i=1;i<=rowIndex;i++){
			res.add(0);
			for(int j=res.size()-1;j>=1;j--){
				res.set(j,res.get(j)+res.get(j-1));
			}
		}
		return res;
	}
}